src/OrchardCore.Modules/OrchardCore.Admin/AdminActionModelConvention.cs

```diff
@@ -0,0 +1,24 @@
+using System;
+using System.Linq;
+using Microsoft.AspNetCore.Mvc.ApplicationModels;
+
+namespace OrchardCore.Admin
+{
+    /// <summary>
+    /// Routing convention to apply admin constraint to mapped admin routes only.
+    /// </summary>
+    public class AdminActionModelConvention : IActionModelConvention
+    {
+        public void Apply(ActionModel action)
+        {
+            if (action.Controller.ControllerName.StartsWith("Admin", StringComparison.OrdinalIgnoreCase) ||
```

Comment:
I think we should use `AdminOptions.AdminUrlPrefix` instead of "Admin"

Reply:
No, this is for controller names